### PR TITLE
Remove Python 2

### PR DESCRIPTION
**Pull request by gnossen**
Please note that this PR will merge into a dev branch, not `master`.

In progress. Not yet ready for review or merge.

### CODE REVIEW

**Comment by srini100** [comment]
You can add "do not merge" label to be on the safe side. Should tests and review checks be enabled on this dev branch?

**Comment by gnossen** [comment]
@srini100 Yes. I'm working on getting those configured. That requires an internal change.

File: tools/run_tests/helper_scripts/build_python.sh [hunk 1]
```diff
@@ -157,11 +157,10 @@ curl https://bootstrap.pypa.io/get-pip.py | $VENV_PYTHON
 # pip-installs the directory specified. Used because on MSYS the vanilla Windows
 # Python gets confused when parsing paths.
 pip_install_dir() {
-  PWD=$(pwd)
-  cd "$1"
+  pushd "$1"
```
Comment (mehrdada) [inline tools/run_tests/helper_scripts/build_python.sh]: The proper way to do this in bash is the following:

```(
cd "$1"
#code
# no cd needed; subshell ends
)```

Comment (gnossen) [reply tools/run_tests/helper_scripts/build_python.sh]: I actually considered using it here, but I wasn't sure if un`export`ed variables would propagate to the child. Glad to see that it does work.

I like the lack of explicit change to the original directory, but the fact that it requires a new subshell isn't ideal for all use cases. I still don't think you could, for example, modify a variable in the parent shell.

**Pull request closed without merging**